flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -0,0 +1,1448 @@
+#endif
+		}
+	}
+	if (halign == -1) {
+		write_hline_lm(xx, xx + dim.width - 1, y - arrow_len, 1, 1);
+		write_hline_lm(xx, xx + dim.width - 1, y + arrow_len - 2, 1, 1);
+		write_vline_lm(xx + dim.width - 1, y - arrow_len, y + arrow_len - 2, 1, 1);
+	} else {
+		write_hline_lm(xx, xx - dim.width - 1, y - arrow_len, 1, 1);
+		write_hline_lm(xx, xx - dim.width - 1, y + arrow_len - 2, 1, 1);
+		write_vline_lm(xx - dim.width - 1, y - arrow_len, y + arrow_len - 2, 1, 1);
+	}
+	// Draw the text.
+	if (halign == -1) {
+		write_string(temp, xx, y, 1, 0, TEXT_VA_MIDDLE, TEXT_HA_LEFT, 0, FONT_ID);
+	} else {
+		write_string(temp, xx, y, 1, 0, TEXT_VA_MIDDLE, TEXT_HA_RIGHT, 0, FONT_ID);
+	}
+#ifdef VERTICAL_SCALE_BRUTE_FORCE_BLANK_OUT
+	// This is a bad brute force method destuctive to other things that maybe drawn underneath like e.g. the artificial horizon:
+	// Then, add a slow cut off on the edges, so the text doesn't sharply
+	// disappear. We simply clear the areas above and below the ticker, and we
+	// use little markers on the edges.
+	if (halign == -1) {
+		write_filled_rectangle_lm(majtick_end + text_x_spacing, y + (height / 2) - (font_info.height / 2), max_text_y - boundtick_start, font_info.height, 0, 0);
+		write_filled_rectangle_lm(majtick_end + text_x_spacing, y - (height / 2) - (font_info.height / 2), max_text_y - boundtick_start, font_info.height, 0, 0);
+	} else {
+		write_filled_rectangle_lm(majtick_end - text_x_spacing - max_text_y, y + (height / 2) - (font_info.height / 2), max_text_y, font_info.height, 0, 0);
+		write_filled_rectangle_lm(majtick_end - text_x_spacing - max_text_y, y - (height / 2) - (font_info.height / 2), max_text_y, font_info.height, 0, 0);
+	}
+#endif
+	y--;
+	write_hline_outlined(boundtick_start, boundtick_end, y + (height / 2), 2, 2, 0, 1);
+	write_hline_outlined(boundtick_start, boundtick_end, y - (height / 2), 2, 2, 0, 1);
+}
+
+/**
+ * hud_draw_compass: Draw a compass.
+ *
+ * @param       v               value for the compass
+ * @param       range           range about value to display (+/- range/2 each direction)
+ * @param       width           length in pixels
+ * @param       x               x displacement
+ * @param       y               y displacement
+ * @param       mintick_step    how often a minor tick is shown
+ * @param       majtick_step    how often a major tick (heading "xx") is shown
+ * @param       mintick_len     minor tick length
+ * @param       majtick_len     major tick length
+ * @param       flags           special flags (see hud.h.)
+ */
+#define COMPASS_SMALL_NUMBER
+// #define COMPASS_FILLED_NUMBER
+void hud_draw_linear_compass(int v, int home_dir, int range, int width, int x, int y, int mintick_step, int majtick_step, int mintick_len, int majtick_len, __attribute__((unused)) int flags)
+{
+	const int FONT_ID = 0;
+	v %= 360; // wrap, just in case.
+	struct FontEntry font_info;
+	int majtick_start = 0, majtick_end = 0, mintick_start = 0, mintick_end = 0, textoffset = 0;
```

Comment:
Hmmm, is this copied from somewhere else? This doesn't look like our normal style. I can't be for sure, but didn't @solidgoldbomb convince us that stylistically we should have every variable be defined on its own line?